src/util/helpers.tsx

```diff
@@ -301,3 +302,21 @@ export const getFileExtension = (filename: string): string => {
 
 export const delay = (ms: number): Promise<void> =>
   new Promise((resolve) => setTimeout(resolve, ms));
+
+export const returnAllDevices = (
+  instance: LxdInstance,
+): [string, LxdDeviceValue][] => {
+  const instanceDevices = Object.entries(instance.expanded_devices ?? {});
+
+  return instanceDevices;
+};
+
+export const getRootPool = (instance: LxdInstance): string => {
+  const allDevices = Object.entries(instance.expanded_devices ?? {}).filter(
+    ([, device]) => {
+      return device.type == "disk" && (device as LxdDiskDevice).path == "/";
+    },
+  );
+  const rootPool = allDevices[0];
+  return (rootPool[1] as LxdDiskDevice).pool;
+};
```

Comment:
```suggestion
export const getRootPool = (instance: LxdInstance): string => {
  const rootStorage = Object.values(instance.expanded_devices ?? {})
    .filter(isDiskDevice)
    .find((device) => {
      return isRootDisk(device as FormDevice);
    });

  return rootStorage ? rootStorage.pool : "";
};
```
A bit safer and a bit easier?

Reply:
There is some code in `InstanceStoragePoolMigration` that is doing this very calculation, I think we should use this function instead of the local `rootDiskDevice` variable in it. Do you mind simplifying that bit as well?

Reply:
I had a look at InstanceStoragePoolMigration and switched it out! Let me know what you think :)